include/git2/hook.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_git_hook_h__
+#define INCLUDE_git_hook_h__
+
+#include "common.h"
+#include "types.h"
+
+/**
+ * @file git2/hook.h
+ * @brief Git hook management routines
+ * @defgroup git_hook Git hook management routines
+ * @ingroup Git
+ * @{
+ */
+GIT_BEGIN_DECL
+
+/**
+ * A helper to get the path of the repository's hooks.
+ *
+ * This will obey core.hooksPath.
+ *
+ * @param out_dir A pointer to a git_buf that will be filled with the absolute path.
+ * @param repo The repository to get the hooks location from.
+ * @return 0 on success, or an error code.
+ */
+GIT_EXTERN(int) git_hook_dir(git_buf *out_dir, git_repository *repo);
+
+typedef int (*git_hook_foreach_cb)(const char *hook_name, void *payload);
```

Comment:
Dumping my brain here as to what might be simpler to use for application developers. When using a callback like this, application developers are required to explicitly know names of possible hooks in order to act on them (assuming he doesn't simply want to know _all_ hooks available). Furthermore, in case the client needs to inspect the hooks, he'll have to assemble each hook's path manually.

I'd propose a revamped API here in that instead of receiving a hook name, we'll introduce an enum `git_hook_type` or similar, which has an enum value for each existing hook. Now the callback will first receive the enum as well as the actual path to the hook so that the client may inspect the file. E.g. something like this:

```C
typedef int (*git_hook_foreach_cb)(git_hook_type type, const char *hook_path, void *payload);
```

Thoughts?